File: frontend/src/services/notificationService.js
import api from "./api";

// Get all notifications
export const getNotifications = async () => {
  const response = await api.get("/api/notifications");
  return response.data;
};

// Get unread count
export const getUnreadCount = async () => {
  const response = await api.get("/api/notifications/unread-count");
  return response.data;
};

// Mark one notification as read
export const markAsRead = async (id) => {
  const response = await api.put(`/api/notifications/${id}/read`);
  return response.data;
};

// Mark all notifications as read
export const markAllAsRead = async () => {
  const response = await api.put("/api/notifications/read-all");
  return response.data;
};

// Delete notification
export const deleteNotification = async (id) => {
  const response = await api.delete(`/api/notifications/${id}`);
  return response.data;
};

File: src/main/java/org/example/digital_warranty/service/NotificationService.java
package org.example.digital_warranty.service;
import org.example.digital_warranty.dto.NotificationResponse;
import org.example.digital_warranty.dto.UnreadCountResponse;
import org.example.digital_warranty.entity.User;

import java.util.List;

public interface NotificationService {

    // Create a notification
    void createNotification(User user, String title, String message);

    // Get all notifications of logged-in user
    List<NotificationResponse> getNotifications();

    // Get unread notification count
    UnreadCountResponse getUnreadCount();

    // Mark a notification as read
    void markAsRead(Long notificationId);

    // Mark all notifications as read
    void markAllAsRead();

    // Delete a notification
    void deleteNotification(Long notificationId);
}